firefox-ios/Client/Frontend/Browser/TermsOfUse/View/TermsOfUseViewController.swift

```diff
@@ -285,8 +285,8 @@ final class TermsOfUseViewController: UIViewController,
         // Only dismiss the view if the tap occurred outside the visible sheetContainer.
         // This prevents dismissing the Terms of Use sheet when interacting with its content.
         if !sheetContainer.frame.contains(sender.location(in: view)) {
-            store.dispatchLegacy(TermsOfUseAction(windowUUID: windowUUID, actionType: .markDismissed))
-            coordinator?.dismissTermsFlow()
+            store.dispatchLegacy(TermsOfUseAction(windowUUID: windowUUID, actionType: .dismissalTimestampSet))
```

Comment:
Do we need to send two events for `gestureDismiss` and `remindMeLaterTapped`? Couldn't we send one and save the timestamp when any of these events are triggered? I didn't see `dismissalTimestampSet` being dispatched in any other context.

Reply:
Indeed, we can prevent sending two events. I updated the code to save the timestamp when any of the two events are triggered. Thank you!